consensus/istanbul/backend/handler.go

```diff
@@ -290,3 +298,191 @@ func (sb *Backend) UnregisterPeer(peer consensus.Peer, isProxiedPeer bool) {
 		}
 	}
 }
+
+// Handshake allows the initiating peer to identify itself as a validator
+func (sb *Backend) Handshake(peer consensus.Peer, peerIsInbound bool) (bool, error) {
```

Comment:
What do you think of using `peer.Inbound()` instead adding a new parameter?  E.g. add that `Inbound` function to the Peer interface.

Reply:
https://github.com/celo-org/celo-blockchain/pull/873/commits/6dff0ee185d9c5408dce55537c18894913e71d05